src/utils/utils.js

```diff
@@ -26,3 +37,59 @@ export function createNewDayCells() {
   }
   return ROW_DATA;
 }
+export function calculateSummaryOfSleep(cells) {
+  const reducer = (map, val) => {
+    if (map[val] == null) {
+      map[val] = 1;
+    } else {
+      ++map[val];
+    }
+    return map;
+  };
+  const result = cells.map((cell) => cell.mode).reduce(reducer, {});
+  return result;
+}
+export function clusterCells(cells) {
+  let currentMode = null;
+  let obj = {
```

Comment:
suggestion: Aim to give good names as much as possible. Here could be `clusterBase`